server/auth_server.go

```diff
@@ -46,40 +46,71 @@ func (a *AuthServer) IsWorkspaceAuthenticated(ctx context.Context, request *api.
 	return &empty.Empty{}, nil
 }
 
+func (a *AuthServer) IsAuthorized(ctx context.Context, request *api.IsAuthorizedRequest) (res *api.IsAuthorizedResponse, err error) {
+	res = &api.IsAuthorizedResponse{}
+	if ctx == nil {
+		res.Authorized = false
+		return res, status.Error(codes.Unauthenticated, "Unauthenticated.")
+	}
+	//User auth check
+	client := ctx.Value("kubeClient").(*v1.Client)
+	_, err2, _ := a.isAuthorized(err, client)
+	if err2 != nil {
+		return nil, err2
+	}
+	//Check the request
+	allowed, err := auth.IsAuthorized(client, request.Namespace, request.Verb, request.Group, request.Resource, request.ResourceName)
+	if err != nil {
+		res.Authorized = false
+		return res, err
+	}
+
+	res.Authorized = allowed
+	return res, nil
+}
+
 func (a *AuthServer) IsValidToken(ctx context.Context, req *api.IsValidTokenRequest) (res *api.IsValidTokenResponse, err error) {
 	if ctx == nil {
 		return nil, status.Error(codes.Unauthenticated, "Unauthenticated.")
 	}
 
 	client := ctx.Value("kubeClient").(*v1.Client)
 
+	_, err2, _ := a.isAuthorized(err, client)
+	if err2 != nil {
+		return nil, err2
+	}
+
+	config, err := client.GetSystemConfig()
+	if err != nil {
+		return
+	}
+	res = &api.IsValidTokenResponse{}
+	res.Domain = config["ONEPANEL_DOMAIN"]
+
+	return res, nil
+}
+
+func (a *AuthServer) isAuthorized(err error, client *v1.Client) (*api.IsValidTokenResponse, error, bool) {
```

Comment:
Bit confused about this, is this combining token validation and authorization?

Reply:
Yes.
My understanding is that the token validation is how we know the user is valid.
The authorization is to see if the user has resource access.

Reply:
Should this be two separate methods instead of one? Or is this intent of this method is for really to check the validity of the token? If so, `isValidToken` might be a better name?

Reply:
The main intent is to validate the user.
They are a valid user, so they can make this request.

I combined the two calls there because that's how it seems we're figuring out the user.

I do agree with the name change, so I made it.